src/osemgrep/cli_scan/Scan_CLI.ml

```diff
@@ -774,7 +785,11 @@ let cmdline_term ~allow_empty_config : conf Term.t =
     in
     let engine_type =
       match (oss, pro_lang, pro_intrafile, pro) with
+      | false, false, false, false when secrets ->
+          Engine_type.(PRO Language_only)
```

Comment:
We should change the Engine_type.ml to use a more precise type.
Actually we might want to get rid of Engine_type.ml and merge with the
engine_kind and product types defined in semgrep_output_v1.atd